test/mail-test-srv/http.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"regexp"
+	"strconv"
+)
+
+// filter is an interface to filter the received mails based on some criteria
+type filter interface {
```

Comment:
What's the purpose of this interface? It looks like it has only one implementation.